packages/trade/exchange/src/pods/matching-engine/matching-engine.service.ts

```diff
@@ -99,6 +89,12 @@ export class MatchingEngineService<TProduct, TProductFilter> implements OnModule
         this.matchingEngine.cancelOrder(orderId);
     }
 
+    public async clear(): Promise<void> {
+        this.matchingEngine.clear();
```

Comment:
Maybe set initialized flag to false so the tick won't get in a way when clearing and reloading, then set to true after all operation completed.